test/compress/screw-ie8.js

```diff
@@ -325,3 +325,61 @@ issue_2120_2: {
     }
     expect_stdout: "PASS"
 }
+
+issue_2254_1: {
+    mangle = {
+        ie8: false,
+    }
+    input: {
+        "eeeeee";
+        try {
+            console.log(f("PASS"));
+        } catch (e) {}
+        function f(s) {
+            try {
+                return s;
+            } catch (e) {}
+        }
+    }
+    expect: {
+        "eeeeee";
+        try {
+            console.log(f("PASS"));
+        } catch (e) {}
+        function f(e) {
+            try {
+                return e;
+            } catch (e) {}
+        }
+    }
+    expect_stdout: "PASS"
+}
+
+issue_2254_2: {
+    mangle = {
+        ie8: true,
+    }
+    input: {
+        "eeeeee";
+        try {
+            console.log(f("PASS"));
+        } catch (e) {}
+        function f(s) {
+            try {
+                return s;
+            } catch (e) {}
+        }
+    }
+    expect: {
+        "eeeeee";
+        try {
+            console.log(f("PASS"));
+        } catch (e) {}
+        function f(c) {
+            try {
+                return c;
+            } catch (e) {}
```

Comment:
> Your example won't work due to try{}catch(e){} being eliminated

https://github.com/mishoo/UglifyJS2/issues/2254#issuecomment-318958809 worked fine. That line wouldn't be eliminated as compress was not run. It was necessary to repro the ie8 bug.

No matter. Your fix works.